components/Newsfeed/Newsfeed.tsx

```diff
@@ -0,0 +1,276 @@
+      <InstantSearch
+        indexName={initialSortByValue}
+        initialUiState={{
+          [initialSortByValue]: {
+            refinementList: { court: [String(currentGeneralCourt)] }
+          }
+        }}
+        searchClient={searchClient}
+        routing={false}
+        // routing={useRouting()}
+      >
+        <Layout />
+      </InstantSearch>
+    </SearchErrorBoundary>
+  )
+}
+
+const Layout = () => {
+  const isMobile = useMediaQuery("(max-width: 768px)")
+
+  const { user } = useAuth()
+  const { result: profile, loading } = usePublicProfile(user?.uid)
+  const uid = user?.uid
+  const subscriptionRef = collection(
+    firestore,
+    `/users/${uid}/userNotificationFeed/`
+  )
+
+  const [orgFilter, setOrgFilter] = useState<boolean>(true)
+  const [billFilter, setBillFilter] = useState<boolean>(true)
+
+  const [allResults, setAllResults] = useState<Props>([])
+  const [orgResults, setOrgResults] = useState<Props>([])
+  const [billResults, setBillResults] = useState<Props>([])
+
+  const [notificationsDisplay, setNotificationsDisplay] = useState<Props>([])
+  const [noDisplay, setNoDisplay] = useState<boolean>(true)
+
+  const onFilterChange = (trolley: string, boxType: string) => {
+    const Box = document.getElementById(boxType) as HTMLInputElement
+    if (Box?.checked) {
+      trolley ? setOrgFilter(true) : setBillFilter(true)
+    } else {
+      trolley ? setOrgFilter(false) : setBillFilter(false)
+    }
+  }
+
+  const checkDisplay = useCallback(() => {
+    setNoDisplay(false)
+    if (orgFilter == true && billFilter == true) {
+      return setNotificationsDisplay(allResults)
+    } else if (orgFilter == true) {
+      return setNotificationsDisplay(orgResults)
+    } else if (billFilter == true) {
+      return setNotificationsDisplay(billResults)
+    }
+    return setNoDisplay(true)
+  }, [allResults, orgFilter, orgResults, billFilter, billResults])
+
+  const notificationQuery = async () => {
+    let notificationList: Props = []
+    const q = query(subscriptionRef, where("uid", "==", `${uid}`))
+    const querySnapshot = await getDocs(q)
+    querySnapshot.forEach(doc => {
+      // doc.data() is never undefined for query doc snapshots
+      notificationList.push(doc.data().notification)
+    })
+
+    if (notificationsDisplay.length === 0 && notificationList.length != 0) {
+      setAllResults(notificationList)
+      setOrgResults(
+        notificationList.filter(notification => notification.type === "org")
+      )
+      setBillResults(
+        notificationList.filter(notification => notification.type === "bill")
+      )
+      setNotificationsDisplay(allResults)
+    }
+  }
+
+  useEffect(() => {
+    uid ? notificationQuery() : null
+  })
+
+  useEffect(() => {
+    checkDisplay()
+  }, [checkDisplay, orgFilter, billFilter, notificationsDisplay])
+
+  function Filters() {
+    return (
+      <FilterBoxs
+        isMobile={isMobile}
+        onFilterChange={onFilterChange}
```

Comment:
```suggestion
        onOrgFilterChange={(isFiltering) => {
          onOrgFilterChange(isFiltering);
          onFilterChange();
        }
        onBillFilterChange={(isFiltering) => {
          onBillFilterChange(isFiltering)
          onFilterChange();
        }
```

Reply:
I needed to fold a variation of onFilterChange into onOrgFilterChange and onBillFilterChange respectively and then make some more changes to handle logic being run on state that's being set; but the state wasn't being set until the end causing the UI to get out of sync with the results.